### ISSUE TITLE
.netcore2.0 got System.Net.Sockets.SocketException: Invalid argument

### ISSUE THREAD

**Issue by benzsuankularb** [statianzo/Fleck#213]

**Comment by statianzo** [comment]
@benzsuankularb any pointers on what is necessary to support standard? Where is Fleck broken?

**Comment by benzsuankularb** [comment]
I got `Initialization method Fleck.Tests.WebSocketServerTests.Setup threw exception. System.Net.Sockets.SocketException: Invalid argument.` when using Fleck with .net core 2.0

After dig down I got the issue with this block
I think .net core still not support SocketOptionLevel.IPv6 SocketOptionName.IPv6Only argument
```
if(!MonoHelper.IsRunningOnMono()){
   #if __MonoCS__
   #else
      socket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.IPv6Only, false);
   #endif
}
```

Do I need to add __MonoCS__ in my proj

**Comment by MatthewFrench** [comment]
I ran into the same error.

**Comment by xPaw** [comment]
I think I have the same problem with net core on Linux:

Which is related to the same `SetSocketOption` line.

```
Unhandled Exception: System.Net.Sockets.SocketException: Protocol not available
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, Int32 optionValue, Boolean silent)
   at System.Net.Sockets.Socket.SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, Int32 optionValue)
   at Fleck.WebSocketServer..ctor(String location)
```

**Comment by quynh-ng** [comment]
Its fixed in the version 1.0.3
